algorithm/cgal/Geometry.h

```diff
@@ -128,17 +142,54 @@ class Geometry {
 
   bool has_input_mesh(int nth) { return input_meshes_[nth] != nullptr; }
 
-  const Surface_mesh& input_mesh(int nth) { return *input_meshes_[nth]; }
+  const Surface_mesh& input_mesh(int nth) {
+#ifdef ENABLE_OCCT
+    if (!has_input_mesh(nth) && has_occt_shape(nth)) {
+      Surface_mesh mesh;
+      buildSurfaceMeshFromOcctShape(occt_shape(nth), /*tolerance=*/0.01, mesh);
+      input_meshes_[nth] = std::shared_ptr<const Surface_mesh>(new Surface_mesh(mesh));
+    }
+#endif
+    return *input_meshes_[nth];
+  }
 
-  bool has_mesh(int nth) { return meshes_[nth] != nullptr; }
+  bool has_mesh(int nth) {
+    return meshes_[nth] != nullptr;
+  }
 
   Surface_mesh& mesh(int nth) {
     if (!has_mesh(nth)) {
       meshes_[nth].reset(new Surface_mesh);
+#ifdef ENABLE_OCCT
+      if (has_occt_shape(nth)) {
+        // TODO: Pick a better default tolerance.
+        buildSurfaceMeshFromOcctShape(occt_shape(nth), /*tolerance=*/0.01, *meshes_[nth]);
```

Comment:
What does the tolerance do here? Also I really like this style of comment, I didn't know you could do that

Reply:
According to https://dev.opencascade.org/doc/refman/html/class_b_rep_mesh___incremental_mesh.html#a3a383b3afe164161a3aa59a492180ac6 it is the linear deflection.

So how far out of true points may be from the ideal curve.